source/val/validate_memory.cpp

```diff
@@ -399,6 +399,18 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
     }
   }
 
+  // 2.18.2. It is invalid to apply both Restrict and Aliased to the same <id>.
```

Comment:
I know this is a wrong place for this piece of code. It needs to be somewhere that we are validating IDs. I found this _spv_result_t spvValidateIDs(const spv_instruction_t* pInstructions,_ in validate.h. Where is the definition?

Reply:
I think this is left over from a refactor. That method was split up into the various ValidateMemory, etc methods.

Reply:
Do I remove it or we'd like to keep there?
Any suggestion about where to place this code?

Reply:
Its dead, go ahead and remove it. 

I would consider having a new function validate_decorations. If you see Aliased, then check other decorations of that id and issue an error if Restrict is one of them.

Alternative: Aliased and Restrict can only be applied to Memory Object Declarations (OpVariable or OpFunctionParameter), so you could add checks in ValidateVariable and ValidateFunctionParameter. This is less ideal since it is probably more code duplication.